### ISSUE TITLE
NPE while using check_mk for monitoring Graylog

### ISSUE THREAD

**Issue by dnauck** [Graylog2/graylog2-server#11350]
Hello,

the check_mk Graylog agent triggers a NPE:

```
2021-09-24T22:04:06.660+02:00 ERROR [AnyExceptionClassMapper] Unhandled exception in REST resource
java.lang.NullPointerException: null
	at org.bson.BasicBSONObject.getLong(BasicBSONObject.java:159) ~[graylog.jar:?]
	at org.graylog2.system.stats.mongo.MongoProbe.mongoStats(MongoProbe.java:176) ~[graylog.jar:?]
	at org.graylog2.system.stats.ClusterStatsService.mongoStats(ClusterStatsService.java:101) ~[graylog.jar:?]
	at org.graylog2.system.stats.ClusterStatsService.clusterStats(ClusterStatsService.java:75) ~[graylog.jar:?]
	at org.graylog2.rest.resources.system.ClusterStatsResource.systemStats(ClusterStatsResource.java:53) ~[graylog.jar:?]
	at sun.reflect.GeneratedMethodAccessor776.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_292]
	at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_292]
	at org.glassfish.jersey.server.model.internal.ResourceMethodInvocationHandlerFactory.lambda$static$0(ResourceMethodInvocationHandlerFactory.java:52) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher$1.run(AbstractJavaResourceMethodDispatcher.java:124) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.invoke(AbstractJavaResourceMethodDispatcher.java:167) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.JavaResourceMethodDispatcherProvider$TypeOutInvoker.doDispatch(JavaResourceMethodDispatcherProvider.java:219) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.internal.AbstractJavaResourceMethodDispatcher.dispatch(AbstractJavaResourceMethodDispatcher.java:79) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.invoke(ResourceMethodInvoker.java:469) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:391) ~[graylog.jar:?]
	at org.glassfish.jersey.server.model.ResourceMethodInvoker.apply(ResourceMethodInvoker.java:80) ~[graylog.jar:?]
	at org.glassfish.jersey.server.ServerRuntime$1.run(ServerRuntime.java:255) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:248) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors$1.call(Errors.java:244) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:292) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:274) [graylog.jar:?]
	at org.glassfish.jersey.internal.Errors.process(Errors.java:244) [graylog.jar:?]
	at org.glassfish.jersey.process.internal.RequestScope.runInScope(RequestScope.java:265) [graylog.jar:?]
	at org.glassfish.jersey.server.ServerRuntime.process(ServerRuntime.java:234) [graylog.jar:?]
	at org.glassfish.jersey.server.ApplicationHandler.handle(ApplicationHandler.java:680) [graylog.jar:?]
	at org.glassfish.jersey.grizzly2.httpserver.GrizzlyHttpContainer.service(GrizzlyHttpContainer.java:356) [graylog.jar:?]
	at org.glassfish.grizzly.http.server.HttpHandler$1.run(HttpHandler.java:200) [graylog.jar:?]
	at com.codahale.metrics.InstrumentedExecutorService$InstrumentedRunnable.run(InstrumentedExecutorService.java:180) [graylog.jar:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_292]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_292]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_292]
```

* Graylog Version: 4.1.5
* Operating System: Ubuntu focal

Check_mk (2.0.0p11): https://checkmk.com/de/integrations?tags=graylog

**Comment by asachs01** [comment]
Hmmmm, what check are you running? AFAIK we don't publish a checkmk agent, so I don't believe this would be a bug that Graylog would need to fix 🤔

**Comment by dnauck** [comment]
I'm using the bundled one from the check_mk distribution (Version 2.0.0p11)

**Comment by gurubert** [comment]
The code that is used for the check can be found here:

https://github.com/tribe29/checkmk/blob/master/cmk/special_agents/agent_graylog.py

**Comment by bernd** [comment]
@dnauck According to the stack trace, there seems to be a problem regarding MongoDB. Which MongoDB version are you using?

**Comment by dnauck** [comment]
I'm using:

```
ii  mongodb-org-server                      4.4.9
```

**Comment by bernd** [comment]
@dnauck Thanks for the info. This is an issue with MongoDB >= 4.4. We have a pull request that will fix the issue: #11357
